src/cdk/table/table.ts

```diff
@@ -879,6 +872,73 @@ export class CdkTable<T> implements AfterContentChecked, CollectionViewer, OnDes
     Array.from(this._columnDefsByName.values()).forEach(def => def.resetStickyChanged());
   }
 
+  /** Invoked whenever an outlet is created and has been assigned to the table. */
+  _outletAssigned(): void {
+    // Trigger the first render once all outlets have been assigned. We do it this way, as
+    // opposed to waiting for the next `ngAfterContentChecked`, because we don't know when
+    // the next change detection will happen.
+    // Also we can't use queries to resolve the outlets, because they're wrapped in a
+    // conditional, so we have to rely on them being assigned via DI.
+    if (
+      !this._hasRendered &&
+      this._rowOutlet &&
+      this._headerRowOutlet &&
+      this._footerRowOutlet &&
+      this._noDataRowOutlet
+    ) {
+      this._hasRendered = true;
+      this._render();
+    }
+  }
+
+  /** Renders the table if its state has changed. */
+  private _render(): void {
```

Comment:
This is a big diff, but I basically just moved the code from `ngAfterContentChecked` into a separate method.